### PR TITLE
Upgrade messages and gherkin to bring new 'jsonschema messages'

### PR DESCRIPTION
**Pull request by aurelien-reeves**

### CODE REVIEW

**Comment by aslakhellesoy** [comment]
FYI the markdown parser is not uet implemented in gherkin/ruby

**Comment by aurelien-reeves** [comment]
Oh, yes, forgot that.
I remembered it when I first create my local branch, but forgot it at the time of creating the PR 😅

File: cucumber-core.gemspec [hunk 1]
```diff
@@ -22,8 +22,8 @@ Gem::Specification.new do |s|
                     'source_code_uri' => 'https://github.com/cucumber/cucumber-ruby-core',
                   }
 
-  s.add_dependency 'cucumber-gherkin', '~> 18.1', '>= 18.1.0'
-  s.add_dependency 'cucumber-messages', '~> 15.0', '>= 15.0.0'
+  s.add_dependency 'cucumber-gherkin', '~> 19.0', '>= 19.0.1'
```
Comment (aslakhellesoy) [inline cucumber-core.gemspec:25]: I've released 19.0.2

**Review by aslakhellesoy — CHANGES_REQUESTED** [review]
Nice one! Ship it :-)

File: cucumber-core.gemspec (hunk 1 above)
Comment (aurelien-reeves) [reply cucumber-core.gemspec:25]: Ok :D

I'll take a look on it tomorrow 👍

**Comment by aurelien-reeves** [comment]
Please do not merge yet.
We may choose another internal API for messages.
See the discussion started here: https://github.com/cucumber/cucumber-ruby/pull/1544#issuecomment-846257924

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,13 @@ Please see [CONTRIBUTING.md](https://github.com/cucumber/cucumber/blob/master/CO
 
 ### Dependencies
 
+* Upgraded to gherkin v20 and messages v17.
+
+  This remove protocol buffer from the dependencies.
```
Comment (luke-hill) [inline CHANGELOG.md:19]: This removes protocol buffers from the runtime dependencies

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,13 @@ Please see [CONTRIBUTING.md](https://github.com/cucumber/cucumber/blob/master/CO
...
+
+  This deeply impact APIs as internal representation of messages has changed from
```
Comment (luke-hill) [inline CHANGELOG.md:21]: This deeply impacts the API as the internal representation ...

**Review by luke-hill — COMMENTED** [review]
Couple of language changes, but basically approved

**Pull request merged**